openfl/federated/data/sources/verifiable_dataset_info.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""This module contains the VerifiableDatasetInfo class."""
+
+import json
+from enum import Enum
+from hashlib import sha384
+from pathlib import Path
+from typing import List
+
+from openfl.federated.data.sources.data_source import DataSource, DataSourceType
+from openfl.federated.data.sources.local_data_source import LocalDataSource
+
+
+class DatasetFormat(Enum):
+    """Enum for the different dataset commitment formats."""
+
+    VERBOSE = "verbose_dataset"
+    CONCISE = "concise_dataset"
+
+
+def filter_non_serializable(obj):
+    """Filter out methods and non-serializable objects."""
+    serializable_dict = {}
+    for key, val in obj.__dict__.items():
+        if callable(val):
+            continue  # Skip methods
```

Comment:
Strange... How would a callable make its way into a dataset source? Am I missing something?

Reply:
because we serialize DataSources class objects, so the default is to include the methods.

Reply:
That makes sense, but then can't we filter them at source (i.e. during the serialization step)?

Reply:
With that function we didn't have to implement serialization in the sources.
However, following your comment I think that it makes more sense to enable each data source to control its serialization,
so now this logic is in DataSource (`_serialize_fields`) and can be overridden in subclasses if needed.